datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -583,25 +588,32 @@ protected void performElasticTaskCountValidation(DatastreamGroupPartitionsMetada
     int totalPartitions = datastreamPartitions.getPartitions().size();
     int numTasksNeeded = (totalPartitions / allowedPartitionsPerTask)
         + (((totalPartitions % allowedPartitionsPerTask) == 0) ? 0 : 1);
+    LOG.info("Number of tasks needed: {}, total task count: {}", numTasksNeeded, totalTaskCount);
```

Comment:
This log will be very useful to debug production issues. Add Datastream information here for debugging. This may not be a big deal while debugging using logs, but with Kusto, this log will not make complete sense. If it is too much to get datastream name info in this, you can summarize all the estimators and final task count in the caller, so that we can use Kusto to look for this log.

Reply:
Not you fault, but looks like there are bunch of places in elastic task count path where datastream group info will be useful in the log.

Reply:
Added datastream group name to the highlighted log. We can tackle other log improvements in a separate pull request.